src/ble/CHIPBleServiceData.h

```diff
@@ -64,17 +67,34 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint8_t GetAdvertisementVersion() const
+    {
+        uint8_t advertisementVersion = static_cast<uint8_t>(DeviceDiscriminatorAndAdvVersion[1] & ~kAdvertisementVersionMask);
+        advertisementVersion         = static_cast<uint8_t>(advertisementVersion >> ShiftBits);
+        return advertisementVersion;
+    }
+
+    // Use only 4 bits to set advertisement version
+    void SetAdvertisementVersion(uint8_t advertisementVersion)
+    {
+        // Advertisement Version is 4 bit long from 12th to 15th
+        advertisementVersion &= kAdvertisementVersionMask;
+        advertisementVersion = static_cast<uint8_t>(advertisementVersion << ShiftBits);
+        DeviceDiscriminatorAndAdvVersion[1] &= kAdvertisementVersionMask;
```

Comment:
This is again relying on this being both the mask for advertisement version and the mask for "not advertisement version".  This is pretty confusing.  We should be using `~kAdvertisementVersionMask` to mean "not advertisement version" and `kAdvertisementVersionMask` to mean "advertisement version".